latch/ldata/path.py

```diff
@@ -0,0 +1,171 @@
+from pathlib import Path
+from typing import Generator, Optional, Union
+from urllib.parse import urljoin
+from uuid import uuid4
+
+import gql
+from latch_sdk_config.latch import NUCLEUS_URL
+from latch_sdk_gql.execute import execute
+
+from latch.ldata.node import (
+    LDataNodeType,
+    LDataPerms,
+    PermLevel,
+    get_node_data,
+    get_node_metadata,
+    get_node_perms,
+)
+from latch.ldata.transfer import download, remote_copy, upload
+from latch.ldata.transfer.progress import Progress
+from latch_cli.tinyrequests import post
+from latch_cli.utils import get_auth_header, urljoins
+from latch_cli.utils.path import is_remote_path
+
+
+class LPath:
+    def __init__(self, path: str):
+        if not is_remote_path(path):
+            raise ValueError(f"Invalid LPath: {path} is not a Latch path")
+        self.path = path
+        self._node_id = None
+
+    @property
+    def node_id(self) -> str:
+        if self._node_id is None:
+            self._node_id = get_node_data(self.path).data[self.path].id
+        return self._node_id
+
+    @property
+    def exists(self) -> bool:
+        try:
+            node_data = get_node_data(self.path).data[self.path]
+        except FileNotFoundError:
+            return False
+        return not node_data.removed
+
+    @property
+    def name(self) -> str:
+        return get_node_data(self.path).data[self.path].name
+
+    @property
+    def type(self) -> LDataNodeType:
+        return get_node_data(self.path).data[self.path].type
+
+    def is_dir(self) -> bool:
+        return self.type in {
+            LDataNodeType.dir,
+            LDataNodeType.account_root,
+            LDataNodeType.mount,
+        }
+
+    @property
+    def size(self) -> float:
+        metadata = get_node_metadata(self.node_id)
+        return metadata.size
+
+    @property
+    def content_type(self) -> str:
+        metadata = get_node_metadata(self.node_id)
+        return metadata.content_type
+
+    def iterdir(self) -> Generator[Path, None, None]:
+        if not self.is_dir():
+            raise ValueError(f"Not a directory: {self.path}")
+        data = execute(
+            gql.gql("""
+            query LDataChildren($argPath: String!) {
+                ldataResolvePathData(argPath: $argPath) {
+                    finalLinkTarget {
+                        childLdataTreeEdges(filter: { child: { removed: { equalTo: false } } }) {
+                            nodes {
+                                child {
+                                    name
+                                }
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"argPath": self.path},
+        )["ldataResolvePathData"]
+
+        if data is None:
+            raise ValueError(f"No directory found at path: {self.path}")
```

Comment:
arguable `FileNotFoundError`? assuming that one is not tied to `OSError` cuz then it'd be the wrong type

and we should probably make our own error class then

---

also canonical python error messages start with lowercase letters:
```suggestion
            raise ValueError(f"no such Latch file or directory: {self.path}")
```

Reply:
agree should be filenotfounderror